pkg/server/domain/service/application.go

```diff
@@ -315,6 +316,35 @@ func (c *applicationServiceImpl) GetApplicationStatus(ctx context.Context, appmo
 	return &app.Status, nil
 }
 
+// GetApplicationStatusFromAllEnvs get applications status from all envs
+func (c *applicationServiceImpl) GetApplicationStatusFromAllEnvs(ctx context.Context, appmodel *model.Application) ([]*apisv1.ApplicationStatusResponse, error) {
+	envBindings, err := c.EnvBindingService.GetEnvBindings(ctx, appmodel)
```

Comment:
Can we reuse the GetApplicationCR function?

Reply:
GetApplicationCR only return all applications that will loss env info.